src/screens/Distribution/index.js

```diff
@@ -0,0 +1,203 @@
+import { compose } from "redux";
+import { connect } from "react-redux";
+
+import { Trans, translate } from "react-i18next";
+import { View, StyleSheet } from "react-native";
+// $FlowFixMe
+import { FlatList, SafeAreaView, ScrollView } from "react-navigation";
+import type { NavigationScreenProp } from "react-navigation";
+import type {
+  Account,
+  CryptoCurrency,
+  Currency,
+  Portfolio,
+} from "@ledgerhq/live-common/lib/types";
+import i18next from "i18next";
+import { getAssetsDistribution } from "@ledgerhq/live-common/lib/portfolio";
+import { createStructuredSelector } from "reselect";
+import { BigNumber } from "bignumber.js";
+import type { AssetsDistribution } from "@ledgerhq/live-common/lib/types/portfolio";
+import type { T } from "../../types/common";
+import TrackScreen from "../../analytics/TrackScreen";
+import { accountsSelector } from "../../reducers/accounts";
+import DistributionCard from "./DistributionCard";
+import LText from "../../components/LText";
+import type { DistributionItem } from "./DistributionCard";
+import {
+  counterValueCurrencySelector,
+  counterValueExchangeSelector,
+  currencySettingsSelector,
+  intermediaryCurrency,
+} from "../../reducers/settings";
+import CounterValues from "../../countervalues";
+import colors from "../../colors";
+import RingChart from "./RingChart";
+import { portfolioSelector } from "../../actions/portfolio";
+import CurrencyUnitValue from "../../components/CurrencyUnitValue";
+
+type Props = {
+  navigation: NavigationScreenProp<*>,
+  currencies: CryptoCurrency[],
+  getCounterValue: (Currency, BigNumber) => *,
+  accounts: Account[],
+  portfolio: Portfolio,
+  counterValueCurrency: Currency,
+  t: T,
+};
+
+const mapStateToProps = createStructuredSelector({
+  getCounterValue: state => {
+    const counterValueCurrency = counterValueCurrencySelector(state);
+    const toExchange = counterValueExchangeSelector(state);
+    return (currency, value) => {
+      const currencySettings = currencySettingsSelector(state, { currency });
+      const fromExchange = currencySettings.exchange;
+      return CounterValues.calculateWithIntermediarySelector(state, {
+        from: currency,
+        fromExchange,
+        intermediary: intermediaryCurrency,
+        toExchange,
+        to: counterValueCurrency,
+        value,
+        disableRounding: true,
+      });
+    };
+  },
+  counterValueCurrency: counterValueCurrencySelector,
+  portfolio: portfolioSelector,
+  accounts: accountsSelector,
+});
+
+class Distribution extends PureComponent<Props, *> {
+  static navigationOptions = {
+    title: i18next.t("distribution.header"),
+    headerLeft: null,
+  };
+
+  renderItem = ({ item, _ }: { item: DistributionItem, index: number }) => (
+    <DistributionCard item={item} />
+  );
+
+  keyExtractor = item => item.id;
+
+  render() {
+    const distribution: AssetsDistribution = getAssetsDistribution(
+      this.props.accounts,
+      this.props.getCounterValue,
+    );
```

Comment:
I suggest to move that directly in the selector code. so basically you don't send the `getCounterValue` function, instead you send the `distribution` object.

later we might implement cache in live-common but we can only benefit it if it's sent as a prop.

Reply:
Nice nice